### PR TITLE
Ported the legacy CSV importer to use the newer interface.

### PR DESCRIPTION
**Pull request by blais**
Note: I don't like this; I really don't like inheritance. I'd like to replace all my
CSV importers by a petl + single-function setup. But I need to port a whole
bunch of them now and this at least allow me to modernize our configurations
without doing a lot of work.

BTW this is achieved using a base class and should remain compatible with existing importers.

### CODE REVIEW

File: beangulp/importers/csv.py [hunk 1]
```diff
@@ -425,3 +432,94 @@ def get_amounts(self, iconfig, row, allow_zero_amounts, parse_amount):
         with special cases, e.g., columns with currency symbols in them.
         """
         return get_amounts(iconfig, row, allow_zero_amounts, parse_amount)
+
+
+# Deprecated. TODO(blais): Remove this eventually (on a major release).
+class Importer(_CsvImporterBase, identifier.IdentifyMixin, filing.FilingMixin):
+    """Importer for CSV files (deprecated version using the ImporterProtocol).
+
+    Note that this is deprecated; use the ImporterNew class instead.
```
Comment (dnicolodi) [inline beangulp/importers/csv.py:441]: I think `ImporterNew` should be `CsvImporter` here, but see below

File: beangulp/importers/csv.py [hunk 2]
```diff
@@ -425,3 +432,94 @@ def get_amounts(self, iconfig, row, allow_zero_amounts, parse_amount):
...
+        kwds.update(prepare_for_identifier(regexps, kwds.get('matchers')))
+        kwds.update(prepare_for_filing(account, kwds.get('prefix', None), institution))
+        super().__init__(config, account, currency, **kwds)
+
+    def extract(self, file, existing_entries=None):
+        account = self.file_account(file)
+        return self._do_extract(file, account, existing_entries)
+
+
+class CsvImporter(importer.Importer):
```
Comment (dnicolodi) [inline beangulp/importers/csv.py:470]: `CSVImporter` maybe as CSV is an acronym?

File: beangulp/importers/csv.py [hunk 3]
```diff
@@ -154,14 +157,41 @@ def normalize_config(config, head, dialect='excel', skip_lines: int = 0):
...
+    """Prepare kwds for filing mixin."""
+    kwds = {'filing': account}
+    if institution:
+        prefix = kwds.get('prefix', None)
+        assert prefix is None
+        kwds['prefix'] = institution
+    return kwds
+
+
+class _CsvImporterBase:
```
Comment (dnicolodi) [inline beangulp/importers/csv.py:184]: `CSVImporterBase` maybe as CSV is an acronym?

**Review by dnicolodi — COMMENTED** [review]
It looks reasonable to me. Just bikeshedding about the class names...

**Comment by dnicolodi** [comment]
Sorry, it seems I messed up somewhere. Let me fix this up and merge it to master.

**Comment by blais** [comment]
Thanks!

**Pull request merged**